### ISSUE TITLE
The `Markdown` widget isn't handling ordered lists correctly

### ISSUE THREAD

**Issue by davep** [Textualize/textual#2002]
In Markdown the number prefix for an item in an ordered list [normally isn't important](https://www.markdownguide.org/basic-syntax/#ordered-lists). Every item can start with any valid numeric value; rendering will have them as an ordered sequence of numbers starting with the number of the first item.

For example:

```markdown
1. One
1. Two
1. Three
1. Four
1. Five
```

comes out as:

1. One
1. Two
1. Three
1. Four
1. Five

However, in our `Markdown` widget at the moment, it appears like this:

<img width="88" alt="Screenshot 2023-03-09 at 13 04 05" src="https://user-images.githubusercontent.com/28237/224031762-23446a25-0849-486b-aeb8-2a6616479fa2.png">

This may be down to [the underlying markdown parser we're using](https://github.com/executablebooks/markdown-it-py). It says it conforms to [the CommonMark specification](https://spec.commonmark.org/), and that specification does state:

> The start number of an ordered list is determined by the list number of its initial list item. The numbers of subsequent list items are disregarded.
> -- <cite>[Section 5.3](https://spec.commonmark.org/0.30/#lists) of [v0.30](https://spec.commonmark.org/0.30/#lists)</cite>

so either we need to do something at our end to do that or enable that, or we may need to raise an issue upstream.

**Comment by davep** [comment]
Trying the [live demo](https://markdown-it.github.io/) it looks like that does the right thing; which would suggest that the handling of this might need to be done in the code that uses the library (I'm not saying that for certain; just that on the surface it suggests that).

**Comment by TomJGooding** [comment]
Rendering html as shown in the live demo would only involve creating the appropriate `<ol>` and `<li>` tags wouldn't it, rather than any actual numbering?

According to the docs for `token.info`, this is "The string value of the item marker for ordered-list “list_item_open” tokens".

So possibly the code here is actually just using the number from the original?

```python
        ...
        for token in parser.parse(markdown):
            ...
            elif token.type == "list_item_open":
                if token.info:
                    stack.append(MarkdownOrderedListItem(f"{token.info}. "))
```

**Comment by davep** [comment]
Yup, I think we'll need to pull out the initial value and do some increment of our own.

**Comment by TomJGooding** [comment]
I suppose an alternative if you wanted to avoid adding your own code to handle this would be to run the markdown through a formatter before trying to parse it?

I haven't read the docs in much detail yet but something like...

```python
import mdformat

unformatted = """
1. One
1. Two
1. Three
1. Four
1. Five
"""

formatted = mdformat.text(
    unformatted,
    options={
        "number": True,
    },
)

print(formatted)
# 1. One
# 2. Two
# 3. Three
# 4. Four
# 5. Five

```
